### PR TITLE
fix:(Chart) fix internalChartHelper mergeChartStyleConfigs function

### PR DESCRIPTION
**Pull request by qxqzx13**

### CODE REVIEW

File: frontend/src/app/utils/internalChartHelper.ts [hunk 1]
```diff
@@ -398,6 +398,48 @@ function getMeta(key, column) {
...
+    case 'number':
+    case 'string':
+      if (typeof sEle?.[v] === 'number' || typeof sEle?.[v] === 'string')
+        return sEle?.[v];
+      break;
+    case 'object':
+      if (tEle[v] === null) {
+        return sEle?.[v];
+      } else if (
+        Object.prototype.toString.call(tEle[v]) === '[object Array]' &&
```
Comment (Cuiyansong) [inline frontend/src/app/utils/internalChartHelper.ts:422]: Array.isArray?

File: frontend/src/app/utils/internalChartHelper.ts [hunk 2]
```diff
@@ -398,6 +398,48 @@ function getMeta(key, column) {
   };
 }
 
+export function determineTheMergeChartStyleConfigsValueType(
```
Comment (Cuiyansong) [inline frontend/src/app/utils/internalChartHelper.ts:401]: if only used in `this` class, it is better to remove `export` key word

File: frontend/src/app/utils/internalChartHelper.ts [hunk 3]
```diff
@@ -398,6 +398,48 @@ function getMeta(key, column) {
...
+  tEle: any,
```
Comment (Cuiyansong) [inline frontend/src/app/utils/internalChartHelper.ts:402]: if input parameters only use the `value`, it is better to pass the value directly, not the whole object?

File: frontend/src/app/utils/internalChartHelper.ts [hunk 4]
```diff
@@ -424,7 +466,7 @@ export function mergeChartStyleConfigs(
       'key' in tEle ? source?.find(s => s?.key === tEle.key) : source?.[index];
 
     if (!isUndefined(sEle?.['value'])) {
-      tEle['value'] = sEle?.['value'];
+      tEle['value'] = determineTheMergeChartStyleConfigsValueType(tEle, sEle);
```
Comment (Cuiyansong) [inline frontend/src/app/utils/internalChartHelper.ts:469]: function name should more clearly, is it `mergeBySourceValueType` ?

**Review by scottsut — APPROVED** [review]

**Pull request merged**